K-tutorial: make several edits

SchmErik opened the pull request:
K-tutorial: Lesson 11: rename isStmt to isStatement

It appears that `isStmt` is a function defined as a part of the BOOL
library. Change the name of this function to that the those who are
working through the tutorial do not write a function that causes
conflicting definitions for `isStmt`.

 K-tutorial: Lesson 04 & Lesson 05: clarify a few phrases

The rewording is purely for clarification and is not intended to change
the over all meaning of the surrounding text.

grosu commented:
added a few more notes.  @daejunpark , probably the issues in this NOTES.md under simple untyped will apply to KOOL as well.

cos commented:
lgtm

daejunpark commented:
lgtm

Commit:
K-tutorial: Lesson 11: rename isStmt to isStatement

It appears that `isStmt` is a function defined as a part of the BOOL
library. Change the name of this function to that the those who are
working through the tutorial do not write a function that causes
confilcting definitions for `isStmt`.

Signed-off-by: Erik Kaneda <erik.kaneda@runtimeverification.com>

Commit:
K-tutorial: clarify a few phrases

The rewording is purely for clarification and is not intended to change
the over all meaning of the surrounding text.

Signed-off-by: Erik Kaneda <erik.kaneda@runtimeverification.com>

ehildenb committed:
Merge branch 'master' into tutorial-edits-00

rv-jenkins committed:
Merge branch 'master' into tutorial-edits-00
